### PR TITLE
Upgrade firestore 3

### PR DESCRIPTION
**Pull request by LightXEthan**
Closes https://github.com/LightXEthan/uwahs-campus-map/issues/96
Fixes https://github.com/LightXEthan/uwahs-campus-map/issues/99 and Fixes https://github.com/LightXEthan/uwahs-campus-map/issues/90

Adds the ability to hold image name and description to poi files (images and audio).
Updated how files are stored in poi and how metadata is handled. Images and audio files are stored as maps that hold the name, url, metaid. The maps are stored in image and audio arrays. 
The metadata of each file will be stored in a separate collection that holds the name, filetype, filepath,(number of references), date added. This is used to delete files (or get description).
This allows easier coding for the reading of photo names/descriptions in the front-end and admin-back. Doc reads are only used for the backend. This allows easy reading for the name of an image without having to search all documents in the files collection for the image’s title each time an image is loaded.
Add a timestamp to the name of a file when uploading a file to make it unique which avoids duplicate files

Data structure 3 - New structure design
- Collection: poi
  - Document: random-id
    - Name: “Barry J Marshall”
    - Location: [31S,115E]
    - Last_modified: last modified
    - Date_added: date added
    - Description
    - Audio: [url_audio1, url_audio2] (to be deleted after implementing below)
    - Image: [url_image1, url_image2] (to be deleted after implementing below)
    - Array: audioArray
      - [0] {name: "", url: "", metaID: ""}
    - Array: imageArray
      - [0] {name: "", url: "", metaID: ""}
      - [1] {name: "", url: "", metaID: ""}

_
- Collection: Files
  - Document: Audio1
    - Name (optional)
    - Description
    - filetype
    - filepath (for deletion)
    - nref (number of times this file is used)
    - date added
  - Document: Image1
    - Name (optional)
    - Description
    - filetype
    - filepath (for deletion)
    - nref (number of times this file is used)
    - date added

### CODE REVIEW

**Comment by LightXEthan** [comment]
Please note that opening poi info screen on the map front that has been created on an old branch won't load.

**Comment by LightXEthan** [comment]
@JKKim98 Ready for review :)

File: src/components/MapPOIInfo/index.js [hunk 1]
```diff
@@ -31,14 +31,16 @@ const MapPOIInfo = props => {
...
-                        alt="UWA History"
-                      />
+                  : poi.imageArray.map(image => (
+                      <div>
+                        <img
+                          src={image.url}
+                          key={image.url.split("token=")[1]}
+                          alt="UWA History"
+                        />
+                      </div>
```
[the remark this answers is not in the document]
Comment (JKKim98) [reply src/components/MapPOIInfo/index.js]: but don't worry about it :D

File: src/components/POIEditForm/index.js [hunk 2]
```diff
@@ -91,81 +93,116 @@ class POIEditForm extends Component {
...
-              .delete();
-          }
-        });
-      })
-      .then(() => {
+  getFileName = () => {
+    if (this.state.isFileOpen) {
+
+    }
+  };
```
[the remark this answers is not in the document]
Comment (LightXEthan) [reply src/components/POIEditForm/index.js]: Forgot to remove this. Thanks! :)

File: src/components/MapPOIInfo/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (LightXEthan) [reply src/components/MapPOIInfo/index.js]: Not sure why I added it but I've reverted it :)

File: src/components/POIEditForm/index.js [hunk 3]
```diff
@@ -91,81 +93,116 @@ class POIEditForm extends Component {
...
+  // Delete the image from imageArray and metafile
+  deleteFile = () => {
+    var metaID = this.state.fileSelected.metaID;
+
+    // Delete the file from the file array, activeTab = 1 when image
+    if (this.state.activeTab === '1') {
+      // Removes the image on the modal
+      let index = this.state.imageArray.indexOf(this.state.fileSelected);
+      if (index > -1) {
+        this.state.imageArray.splice(index, 1);
```
[the remark this answers is not in the document]
Comment (LightXEthan) [reply src/components/POIEditForm/index.js]: Changed in new commit

File: src/components/POIEditForm/index.js [hunk 4]
```diff
@@ -91,81 +93,116 @@ class POIEditForm extends Component {
...
     this.props.firebase.poiDelete(this.props.poi._id);
     this.toggleNestedModal();
     this.toggleModal();
   };
 
+  // Add new file to the current viewing edit for and to firestore
+  addFile = (type, filedata) => {
+    if (type === 'image') {
+      // Add to currently viewing edit form
+      this.state.imageArray.push(filedata);
```
[the remark this answers is not in the document]
Comment (LightXEthan) [reply src/components/POIEditForm/index.js]: Changed in new commit

File: src/components/POIEditForm/index.js [hunk 5]
```diff
@@ -249,25 +295,24 @@ class POIEditForm extends Component {
...
+      return this.state.imageArray.map(image => (
         // key of the image is the url token generated by firebase storage
-        <Col md="6" className="img-col" key={image.split("token=")[1]}>
+        <Col md="6" className="img-col" key={image.url.split("token=")[1]}>
           <Card>
             <CardImg
               className="card-img-top"
-              src={image}
+              src={image.url}
               alt="Image name here"
```
Comment (Elenaires) [inline src/components/POIEditForm/index.js]: Hi @LightXEthan , can you update the alt to {image.name} as well? :)

**Pull request merged**